src/System.Runtime.Extensions/src/System/Environment.cs

```diff
@@ -44,7 +45,15 @@ private static Hashtable ToHashtable(this IEnumerable<KeyValuePair<string, strin
             Hashtable hashTable = new Hashtable();
             foreach (KeyValuePair<string, string> pair in pairs)
             {
-                hashTable.Add(pair.Key, pair.Value);
+                try 
+                {
+                    hashTable.Add(pair.Key, pair.Value);                 
+                }
+                catch (Exception ex)
```

Comment:
Why Exception and not InvalidOperationException (or whatever)? We should always catch the most specific type possible.

Reply:
Sure. I'll switch to ArgumentException